crates/edr_provider/src/data.rs

```diff
@@ -480,13 +613,93 @@ impl ProviderData {
             .map_err(ProviderError::Blockchain)
     }
 
+    pub fn set_min_gas_price(&mut self, min_gas_price: U256) -> Result<(), ProviderError> {
+        if self.spec_id() >= SpecId::LONDON {
+            return Err(ProviderError::SetMinGasPriceUnsupported);
+        }
+
+        self.min_gas_price = min_gas_price;
+
+        Ok(())
+    }
+
     pub fn send_transaction(
         &mut self,
-        transaction_request: EthTransactionRequest,
+        transaction_request: TransactionRequestAndSender,
     ) -> Result<B256, ProviderError> {
-        let signed_transaction = self.sign_transaction_request(transaction_request)?;
+        let signed_transaction = if self.is_auto_mining {
```

Comment:
I noticed in an earlier version of the code we had a chain id check here. Wondering about the reason for removing it? Is the check happening elsewhere?